cibuildwheel/linux.py

```diff
@@ -104,10 +105,12 @@ def build(options: BuildOptions) -> None:
         ('pp', 'manylinux_x86_64', options.manylinux_images['pypy_x86_64']),
     ]
 
-    if not os.path.realpath(options.package_dir).startswith(os.path.realpath('.')):
+    pwd = Path().resolve()
+    abs_package_dir = options.package_dir.resolve()
+    if pwd != abs_package_dir and pwd not in abs_package_dir.parents:
         raise Exception('package_dir must be inside the working directory')
 
-    container_package_dir = os.path.join('/project', os.path.relpath(options.package_dir, '.'))
+    container_package_dir = Path('/project') / abs_package_dir.relative_to(pwd)
```

Comment:
Might be worth using a PurePath here, since this path does not represent anything accessible on this machine.

Reply:
Yes. Might be a good idea to try using that be default and only switch to `Path` when necessary.